client/async_int_test.go

```diff
@@ -39,14 +40,59 @@ var _ = Describe("Client", func() {
 		}
 	})
 
-	Describe("Async", func() {
+	Describe("Not mocking producer", func() {
+		var (
+			consumer *extensions.Consumer
+		)
+
+		BeforeEach(func() {
+			var err error
+			consumer, err = extensions.NewConsumer(config, logger)
+			Expect(err).NotTo(HaveOccurred())
+			go consumer.ConsumeLoop()
+			time.Sleep(1 * time.Second) // wait consumer receive assign partition
+		})
+
+		AfterEach(func() {
+			consumer.Cleanup()
+		})
+
+		startAppAndClient := func() {
+			var err error
+			a, err = app.NewApp("0.0.0.0", 5000, logger, config)
+			Expect(err).NotTo(HaveOccurred())
+			go a.Run()
+			config.Set("client.async", true)
+			c, err = client.NewClient(
+				"",
+				config,
+				logger,
+				nil,
+				grpc.WithDefaultCallOptions(grpc.WaitForReady(true)),
+			)
+			Expect(err).NotTo(HaveOccurred())
+		}
+
+		Describe("Send", func() {
+			It("Should send event asynchronously", func() {
+				startAppAndClient()
+				err := c.Send(context.Background(), name, props)
+				Expect(err).NotTo(HaveOccurred())
+				err = c.GracefulStop()
+				Expect(err).NotTo(HaveOccurred())
+				<-*consumer.MessagesChannel()
```

Comment:
Wrap this in a select with a timeout? If the send doesn't happen the test will block indefinitely.

```go
select {
case <-*consumer.MessagesChannel():
    // assert on the message?
case <-time.Timer(1*time.Second):
    t.Error("timed out waiting for message")
}
```

(out of scope) btw, I noticed consumer.MessagesChannel() returns a pointer to channel. There's little reason to do that, it should just return the channel.

Reply:
done. You're right!